pkg/skaffold/runner/new.go

```diff
@@ -186,7 +186,10 @@ func isImageLocal(runCtx *runcontext.RunContext, imageName string) (bool, error)
 
 	cl := runCtx.GetCluster()
 	var pushImages bool
-	if pipeline.Build.LocalBuild.Push == nil {
+	if runCtx.Opts.PushImages {
```

Comment:
This is unable to distinguish if the `false` flag value was set by default or provided by the user. As such running `skaffold build --push=false` will be the same as running `skaffold build`.  I don't think this works for the linked issue as is.

Reply:
I definitely agree, the UX for the `--push=false` case is not great for users and should not be merged as is.  Any opinions on how to detect if a flag value was set by default or provided by the user?  From our slack discussion there were these options (thanks @MarlonGamez for these suggestions!):
- Implement `BoolOrUndefined` type (similar to `StringOrUndefined` type)
-  Look up flags after creation and check the flag.Changed field.  This might not be possible (or be difficutl) as you would need access to the original cobra.Cmd to lookup flags on it

Reply:
What was the resolution here?

Reply:
I think we can go with `BoolOrUndefined`. That seems like the simplest approach and the precedence is already set with `StringOrUndefined`.

Reply:
Updated the code to add a BoolOrUndefined type that is now used for the `--push` flag

Reply:
Currently the syntax required here is:
--push=true
vs
--push

Not sure if this is an issue for merging, looking at how this might be fixed atm.

Reply:
Check out `NoOptDefVal`

Reply:
Great, did exactly what I needed!  Thanks!